keras_cv/layers/preprocessing/base_image_augmentation_layer.py

```diff
@@ -129,12 +145,101 @@ def __init__(self):
     def auto_vectorize(self, auto_vectorize):
         self._auto_vectorize = auto_vectorize
 
-    @property
-    def _map_fn(self):
+    def compute_image_signature(self, images):
+        """Computes the output image signature for the `augment_image()` function.
+
+        Must be overridden to return tensors with different shapes than the input
+        images.  By default returns either a `tf.RaggedTensorSpec` matching the input
+        image spec, or a `tf.TensorSpec` matching the input image spec.
+        """
+        if self.force_output_dense_images:
+            return tf.TensorSpec(images.shape[1:], self.compute_dtype)
+        if self.force_output_ragged_images or isinstance(images, tf.RaggedTensor):
+            ragged_spec = tf.RaggedTensorSpec(
+                shape=images.shape[1:],
+                ragged_rank=1,
+                dtype=self.compute_dtype,
+            )
+            return ragged_spec
+        return tf.TensorSpec(images.shape[1:], self.compute_dtype)
+
+    # TODO(lukewood): promote to user facing API if needed
+    def _compute_bounding_box_signature(self, bounding_boxes):
```

Comment:
Do I understand correctly that this is forcing bounding boxes to be ragged? If so -- is that too coercive of an API?

Reply:
Yeah, so right now this is always making bounding boxes Ragged.

I'm on the fence - on one hand we can minimize API drift by requiring bounding boxes to be RaggedTensors.  on the other hand it is a bit more coercive.  

We can also internally always process raggeds, and then handle outputs being converted to Dense tensors in `format_output` and preserve the input format.  The issue with this is that if we just keep converting back and forth we're not netting ANY performance gains; its a performance hit.  

I say we submit this as is, and if users request dense bounding box tensors we implement it then.   Thoughts?

Reply:
I'm really not sure what the right answer here is. Maybe @martin-gorner has an opinion?

Reply:
Lets add an API for this if users ask for it.  We'll be processing Raggeds internally no matter what.